tiledb/sm/c_api/tiledb.cc

```diff
@@ -567,17 +567,40 @@ int32_t tiledb_buffer_get_type(
   return TILEDB_OK;
 }
 
-int32_t tiledb_buffer_get_size(
-    tiledb_ctx_t* ctx, const tiledb_buffer_t* buffer, uint64_t* num_bytes) {
+int32_t tiledb_buffer_get_buffer(
+    tiledb_ctx_t* ctx,
+    const tiledb_buffer_t* buffer,
+    void** data,
+    uint64_t* num_bytes) {
   if (sanity_check(ctx) == TILEDB_ERR ||
       sanity_check(ctx, buffer) == TILEDB_ERR)
     return TILEDB_ERR;
 
+  *data = buffer->buffer_->data();
   *num_bytes = buffer->buffer_->size();
 
   return TILEDB_OK;
 }
 
+int32_t tiledb_buffer_set_buffer(
```

Comment:
Minor nit: name recursion. Not sure if these would be any better, but quick brainstorm FWIW: `tiledb_buffer_wrap`, `tiledb_buffer_set_backing` (obscure?), `tiledb_buffer_wrap_existing`, `tiledb_buffer_set_data`, `tiledb_buffer_attach`

Reply:
I like the 'wrap' idea. What about for the getter?

Reply:
`_set_data`, `_get_data` seems intuitive

Reply:
> I like the 'wrap' idea. What about for the getter?

could do `tiledb_buffer_data`, but I guess we tend to have verb+noun in the API, so 👍 for `_{set/get}_data`